overlord/state/state.go

```diff
@@ -363,11 +364,16 @@ func (s *State) tasksIn(tids []string) []*Task {
 // Prune removes changes that became ready for more than pruneWait
 // and aborts tasks spawned for more than abortWait.
 // It also removes tasks unlinked to changes after pruneWait.
-func (s *State) Prune(pruneWait, abortWait time.Duration) {
+func (s *State) Prune(pruneWait, abortWait time.Duration, maxReadyChanges int) {
```

Comment:
"maxReadyChanges" seems misleading since the comparison below looks at the total count instead of just the ready changes. This should probably be "pruneReadyAfter" or similar, and the documentation above should be updated to mention it.